pkg/database/database.go

```diff
@@ -428,22 +439,33 @@ func checkAndMigrateDB(db *DbInstance, migrate bool) error {
 
 			case "1ce8669664d2":
 				log.Info("Migrating database to FastTrackML schema 5d042539be4f")
-				if err := db.Transaction(func(tx *gorm.DB) error {
-					constraints := []string{"Tags", "Runs"}
-					for _, constraint := range constraints {
-						if err := tx.Migrator().DropConstraint(&Experiment{}, constraint); err != nil {
-							return err
-						}
-						if err := tx.Migrator().CreateConstraint(&Experiment{}, constraint); err != nil {
-							return err
+				// We need to run this migration without foreign key constraints to avoid
+				// the cascading delete to kick in and delete all the run data.
+				if err := runWithoutForeignKeyIfPossible(func() error {
```

Comment:
this is crazy, dropping the constraint causes a delete-cascade??

Reply:
most table alterations in SQLite are not possible and thus require recreating the table — if you recreate a table that is referenced by other tables with `ON DELETE CASCADE`, then their content will be deleted 😭

Reply:
that's rough, thanks for catching!

Reply:
so this means we cannot migrate sqlite to have delete-cascades on those associations, yes? Do we need to reopen that ticket and manually clear associations?

Reply:
no no, the `CreateConstraint` is enough to modify the constraint (hence why we skip the `DropConstraint` in SQLite and avoid recreating twice.
and we _do_ want to have `ON DELETE CASCADE` on those associations, but we want to disable it temporarily whilst migrating some tables.